sys/kern/syscalls.c

```diff
@@ -432,12 +432,10 @@ static int sys_pipe2(proc_t *p, pipe2_args_t *args, register_t *res) {
   int error;
 
   klog("pipe2(%x, %x)", u_fdp, flags);
-
-  if (flags & ~O_CLOEXEC) {
+  if (flags & ~O_CLOEXEC & ~O_NONBLOCK) {
```

Comment:
Please add `( )` - I can guarantee you that most of us don't know binop priorities.